sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/SocketClient.java

```diff
@@ -0,0 +1,159 @@
+import com.azure.core.util.BinaryData;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.ProtocolException;
+import java.net.Socket;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * A socket client used for making PATCH requests
+ */
+
+class SocketClient {
+
+    private static final String HTTP_VERSION = " HTTP/1.1";
+
+    /**
+     * Opens a socket connection, then writes the PATCH request across the
+     * connection and reads the response
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @return an instance of HttpUrlConnectionResponse
+     */
+    public static HttpUrlConnectionResponse sendPatchRequest(HttpRequest httpRequest) throws IOException {
+        String host = httpRequest.getUrl().getHost();
+        int port = httpRequest.getUrl().getPort();
+        if (httpRequest.getUrl().getProtocol().equals("https")) {
+            SSLSocketFactory sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            try (SSLSocket socket = (SSLSocket) sslSocketFactory.createSocket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        } else if (httpRequest.getUrl().getProtocol().equals("http")) {
+            try (Socket socket = new Socket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        }
+        throw new ProtocolException("Only HTTP and HTTPS are supported by this client.");
+    }
+
+    /**
+     * Calls buildAndSend to send a String representation of the request across the output
+     * stream, then calls buildResponse to get an instance of HttpUrlConnectionResponse
+     * from the input stream
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param socket An instance of the SocketClient
+     * @return an instance of HttpUrlConnectionResponse
+     */
+    private static HttpUrlConnectionResponse doInputOutput(HttpRequest httpRequest, Socket socket) throws IOException {
+        httpRequest.setHeader(HttpHeaderName.HOST, httpRequest.getUrl().getHost());
+        if (!"keep-alive".equalsIgnoreCase(httpRequest.getHeaders().getValue(HttpHeaderName.CONNECTION))) {
+            httpRequest.setHeader(HttpHeaderName.CONNECTION, "close");
+        }
+
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
+             OutputStreamWriter out = new OutputStreamWriter(socket.getOutputStream())) {
+
+            buildAndSend(httpRequest, out);
+            HttpUrlConnectionResponse response = buildResponse(httpRequest, in);
+
+            String redirectLocation = response.getHeaders().stream()
+                .filter(h -> h.getName().equals("Location"))
```

Comment:
HTTP header names are case insensitive. This means sometimes the header might be `location`, `lOCaTioN`, or any other permutation. You should be case insensitive here too to ensure you don't miss out on a redirection.